v0/destinations/hs/transform.js

```diff
@@ -1,251 +1,704 @@
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+};
+
+const batchEvents = destEvents => {
+  let batchedResponseList = [];
+  const trackResponseList = [];
+  // create contact chunck
+  let createContactEventsChunk = [];
+  // update contact chunk
+  let updateContactEventsChunk = [];
+  // general indentify event chunk
+  let eventsChunk = [];
+  const arrayChunksIdentifyCreateContact = [];
+  const arrayChunksIdentifyUpdateContact = [];
+  const arrayChunksIdentify = [];
+  destEvents.forEach((event, index) => {
+    // handler for track call
+    // track call does not have batch endpoint
+    if (event.message.endpoint.includes("/events/v3/send")) {
+      const { message, metadata, destination } = event;
+      const endpoint = get(message, "endpoint");
+
+      const batchedResponse = defaultBatchRequestConfig();
+      batchedResponse.batchedRequest.headers = message.headers;
+      batchedResponse.batchedRequest.endpoint = endpoint;
+      batchedResponse.batchedRequest.body = message.body;
+      batchedResponse.batchedRequest.params = message.params;
+      batchedResponse.batchedRequest.method =
+        defaultGetRequestConfig.requestMethod;
+      batchedResponse.metadata = [metadata];
+      batchedResponse.destination = destination;
+
+      trackResponseList.push(
+        getSuccessRespEvents(
+          batchedResponse.batchedRequest,
+          batchedResponse.metadata,
+          batchedResponse.destination
+        )
+      );
+    } else if (
+      event.message.endpoint ===
+      "https://api.hubapi.com/crm/v3/objects/contacts"
+    ) {
+      // Identify: making chunks for CRM create contact endpoint
+      createContactEventsChunk.push(event);
+    } else if (
+      event.message.endpoint ===
+      "https://api.hubapi.com/crm/v3/objects/contacts/"
+    ) {
+      // Identify: making chunks for CRM update contact endpoint
+      updateContactEventsChunk.push(event);
+    } else {
+      eventsChunk.push(event);
+    }
+
+    // CRM create contact endpoint chunks
+    if (
+      createContactEventsChunk.length &&
+      (createContactEventsChunk.length === MAX_BATCH_SIZE_CRM_CONTACT ||
+        index === destEvents.length - 1)
+    ) {
+      arrayChunksIdentifyCreateContact.push(createContactEventsChunk);
+      createContactEventsChunk = [];
+    }
+
+    // CRM update contact endpoint chunks
+    if (
+      updateContactEventsChunk.length &&
+      (updateContactEventsChunk.length === MAX_BATCH_SIZE_CRM_CONTACT ||
+        index === destEvents.length - 1)
+    ) {
+      arrayChunksIdentifyUpdateContact.push(updateContactEventsChunk);
+      updateContactEventsChunk = [];
+    }
+
+    // general identify chunks
+    if (
+      eventsChunk.length &&
+      (eventsChunk.length === MAX_BATCH_SIZE_CRM_OBJECT ||
+        index === destEvents.length - 1)
+    ) {
+      arrayChunksIdentify.push(eventsChunk);
+      eventsChunk = [];
+    }
```

Comment:
You can use the lodash function ` _.chunk` here, for reference you can look at mailchimp or snapchat conversions implementation